js/src/main/scala/com/lynbrookrobotics/funkydashboard/Dataset.scala

```diff
@@ -7,13 +7,20 @@ object Dataset {
   def extract(definition: DatasetDefinition): List[(Double, String)] => ReactComponentU[_, _, _, TopNode] = {
     definition match {
       case DatasetDefinition(_, "time-series-numeric") =>
-        values => TimeSeriesNumeric(
-          values.map(v => (v._1, read[TimeSeriesValue](v._2)))
-        )
+        values =>
+          TimeSeriesNumeric(
+            values.map(v => (v._1, read[TimeSeriesValue](v._2)))
+          )
+      case DatasetDefinition(_, "time-snapshot") =>
+        values =>
+          TimeSnapshotNumeric(
+            values.map(v => (v._1, read[TimeSnapshotValue](v._2)))
+          )
       case DatasetDefinition(_, "image-stream") =>
-        values => ImageStream(
-          values.map(v => read[ImageStreamValue](v._2))
-        )
+        values =>
+          ImageStream(
```

Comment:
Can you return this, and the above to the original formatting? We want to minimize diffs.